Added a function to replace bitmap data in BitmapRegistry

charlenni opened the pull request:
Added some comment

pauldendulk commented on Mapsui/Styles/BitmapRegistry.cs:
```diff
@@ -3,42 +3,89 @@
+            if (bitmapData == null)
+                throw new ArgumentException("The bitmap data that is registered is null. Was the image loaded correctly?");
+
+            if (bitmapData is Sprite sprite)
```
Good to do these checks to catch errors early. Unfortunate we have to cast in the blind. Perhaps we should start to work with a bitmap type at some point.

charlenni replied:
Yes, but than we move the blind cast to this bitmap type. We could create a registry function separat for each type like RegisterBitmap, RegisterSVG, RegisterSprite and so on.

pauldendulk commented on Mapsui/Styles/BitmapRegistry.cs:
```diff
@@ -3,42 +3,89 @@
+        /// <param name="id">Id of existing bitmap data</param>
+        /// <param name="bitmapData">New bitmap data to replace</param>
+        /// <returns>True, if replacing worked correct</returns>
+        public bool Set(int id, object bitmapData)
```
Was thinking about 'Replace' as a name. Would be clear. 'Set' is also nice though. 'Update' would be another alternative. Let's keep 'Set'.

charlenni replied:
I thought about all three names, but we have already a Get function, so I took Set as the counterpart.

charlenni commented:
> Okay. Good to check for null and Sprite.Atlas early but I don't think this should be done inside a cache that has so far no knowledge of what is being cached. Good enough for now.

The null check was already part of this function when I started with Mapsui. I added the sprite check when I implemented sprites.

charlenni commented:
Ok, this wouldn't work, because the Skia renderer has its own cache mechanism and so, the replaced bitmap in BitmapRegistry would be draw, but the old in the Skia SymbolCache. :-(
